redis_cluster.c

```diff
@@ -43,6 +43,119 @@ zend_class_entry *redis_cluster_exception_ce;
 #include "redis_cluster_arginfo.h"
 #endif
 
+static void
+cluster_enqueue_response(redisCluster *c, short slot, cluster_cb cb, void *ctx)
+{
+    clusterFoldItem *item;
+
+    item = emalloc(sizeof(clusterFoldItem));
+    item->callback = cb;
+    item->slot = slot;
+    item->ctx = ctx;
+    item->next = NULL;
+    item->flags = c->flags->flags;
+
+    if (UNEXPECTED(c->multi_head == NULL)) {
+        c->multi_head = item;
+        c->multi_curr = item;
+    } else {
+        c->multi_curr->next = item;
+        c->multi_curr = item;
+    }
+}
+
+static void cluster_free_queue(redisCluster *c) {
+    clusterFoldItem *item = c->multi_head, *tmp;
+
+    while (item) {
+        tmp = item->next;
+        efree(item);
+        item = tmp;
+    }
+
+    c->multi_head = NULL;
+    c->multi_curr = NULL;
+}
+
+static void cluster_reset_multi(redisCluster *c) {
+    redisClusterNode *node;
+    ZEND_HASH_FOREACH_PTR(c->nodes, node) {
+        if (node == NULL)
+            break;
```

Comment:
```suggestion
            continue;
```

Reply:
Yeah continue is better here. I saw this too when changing it from the macro which currently calls break. My guess is this is just a bug that has never happened because we never add a null node?